htdocs/modulebuilder/template/myobject_list.php

```diff
@@ -145,8 +145,16 @@
 		$search[$key] = GETPOST('search_'.$key, 'alpha');
 	}
 	if (preg_match('/^(date|timestamp|datetime)/', $val['type'])) {
-		$search[$key.'_dtstart'] = dol_mktime(0, 0, 0, GETPOST('search_'.$key.'_dtstartmonth', 'int'), GETPOST('search_'.$key.'_dtstartday', 'int'), GETPOST('search_'.$key.'_dtstartyear', 'int'));
-		$search[$key.'_dtend'] = dol_mktime(23, 59, 59, GETPOST('search_'.$key.'_dtendmonth', 'int'), GETPOST('search_'.$key.'_dtendday', 'int'), GETPOST('search_'.$key.'_dtendyear', 'int'));
+		if (empty(GETPOST('search_'.$key.'_dtstart', 'int'))) {
+			$search[$key.'_dtstart'] = dol_mktime(0, 0, 0, GETPOST('search_'.$key.'_dtstartmonth', 'int'), GETPOST('search_'.$key.'_dtstartday', 'int'), GETPOST('search_'.$key.'_dtstartyear', 'int'));
```

Comment:
GETPOST('search_'.$key.'_dtstartmonth', 'int') must be used in priority before GETPOST('search_'.$key.'_dtstart', 'int')
In fact, we should never use param  'search_'.$key.'_dtstart'  to make a search on a date but we should send the 3 component of the date  so  'search_'.$key.'_dtstartday', 'search_'.$key.'_dtstartmonth', 'search_'.$key.'_dtstartyear'

In which cas, do you receive _dtstart and not the 3 detailed component ?

Reply:
We do not have the 3 detailed components. But we can get them if we add these lines when filling $param before the final else:

elseif (preg_match('/_dtstart/', $key) && !empty($val)) {
        $param .= '&search_'.$key.'month='.urlencode(GETPOST('search_'.$key.'month', 'int'));
        $param .= '&search_'.$key.'day='.urlencode(GETPOST('search_'.$key.'day', 'int'));
        $param .= '&search_'.$key.'year='.urlencode(GETPOST('search_'.$key.'year', 'int'));
}  elseif (preg_match('/_dtend/', $key) && !empty($val)) {
        $param .= '&search_'.$key.'month='.urlencode(GETPOST('search_'.$key.'month', 'int'));
        $param .= '&search_'.$key.'day='.urlencode(GETPOST('search_'.$key.'day', 'int'));
        $param .= '&search_'.$key.'year='.urlencode(GETPOST('search_'.$key.'year', 'int'));
}

I've tried it and it works, but I don't change the pull for you to choose the best option.

Reply:
Yes please provide the 3 components. the string date is not reliable due to a lot of differences in syntax between languages.

Reply:
Done!